### ISSUE TITLE
ERROR when trying to trade BTC/USDT with stake_currency USDT or BTC 

### ISSUE THREAD

**Issue by masher91** [freqtrade/freqtrade#1098]
## Step 1: Have you search for this issue before posting it?

If you have discovered a bug in the bot, please [search our issue tracker]
New Issue 

## Step 2: Describe your environment

  * Python Version: Python 3.6.3 (`python -V`)
  * CCXT version: ccxt==1.15.42 (`pip freeze | grep ccxt`)
  * Branch: Master | Develop
  * Last Commit ID: _____ (`git log --format="%H" -n 1`)
  
## Step 3: Describe the problem:

I am trying to trade BTC/USDT as the whitelisted currency. 

When I set the stake value as BTC the error I get is....

$ python3 ./freqtrade/main.py -c config.json
2018-07-30 16:32:43,331 - freqtrade.configuration - INFO - Log level set to INFO
2018-07-30 16:32:43,331 - freqtrade.configuration - INFO - Dry run is disabled
2018-07-30 16:32:43,331 - freqtrade.configuration - INFO - Using DB: "sqlite:///tradesv3.sqlite"
2018-07-30 16:32:43,331 - freqtrade.configuration - INFO - Using max_open_trades: 10 ...
2018-07-30 16:32:43,332 - freqtrade.configuration - INFO - Using data folder: user_data/data/bittrex ...
2018-07-30 16:32:43,332 - freqtrade.freqtradebot - INFO - Starting freqtrade 0.17.1
2018-07-30 16:32:43,718 - freqtrade.strategy.resolver - INFO - Using resolved strategy DefaultStrategy from '/Users/matthewsmalbach/freqtrade/freqtrade/strategy'
2018-07-30 16:32:43,718 - freqtrade.strategy.resolver - INFO - Override strategy 'ticker_interval' with value in config file: 1m.
2018-07-30 16:32:43,742 - freqtrade.exchange - INFO - Using Exchange "Bittrex"
2018-07-30 16:32:45,281 - freqtrade - ERROR - Pair BTC/USDT not compatible with stake_currency: BTC

When I run the same config.json with BTC as the stake_currency I get.....

$ python3 ./freqtrade/main.py -c config.json
2018-07-30 16:38:55,941 - freqtrade.configuration - INFO - Log level set to INFO
2018-07-30 16:38:55,942 - freqtrade.configuration - INFO - Dry run is disabled
2018-07-30 16:38:55,942 - freqtrade.configuration - INFO - Using DB: "sqlite:///tradesv3.sqlite"
2018-07-30 16:38:55,942 - freqtrade.configuration - INFO - Using max_open_trades: 10 ...
2018-07-30 16:38:55,943 - freqtrade.configuration - INFO - Using data folder: user_data/data/bittrex ...
2018-07-30 16:38:55,943 - freqtrade.freqtradebot - INFO - Starting freqtrade 0.17.1
2018-07-30 16:38:56,343 - freqtrade.strategy.resolver - INFO - Using resolved strategy DefaultStrategy from '/Users/matthewsmalbach/freqtrade/freqtrade/strategy'
2018-07-30 16:38:56,343 - freqtrade.strategy.resolver - INFO - Override strategy 'ticker_interval' with value in config file: 1m.
2018-07-30 16:38:56,529 - freqtrade.exchange - INFO - Using Exchange "Bittrex"
2018-07-30 16:38:58,082 - freqtrade.persistence - INFO - trying trades_bak0
2018-07-30 16:38:58,083 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: *Status:* `running`
2018-07-30 16:38:58,083 - freqtrade.freqtradebot - INFO - Changing state to: RUNNING
2018-07-30 16:39:01,366 - freqtrade.exchange - WARNING - get_balance() returned exception: "Could not get USDT balance due to malformed exchange response: {'BTC': {'free': 0.62956276, 'used': 0.0, 'total': 0.62956276}}"
2018-07-30 16:39:01,366 - freqtrade.exchange - WARNING - retrying get_balance() still for 3 times
2018-07-30 16:39:02,970 - freqtrade.exchange - WARNING - get_balance() returned exception: "Could not get USDT balance due to malformed exchange response: {'BTC': {'free': 0.62956276, 'used': 0.0, 'total': 0.62956276}}"
2018-07-30 16:39:02,970 - freqtrade.exchange - WARNING - retrying get_balance() still for 2 times
2018-07-30 16:39:04,402 - freqtrade.exchange - WARNING - get_balance() returned exception: "Could not get USDT balance due to malformed exchange response: {'BTC': {'free': 0.62956276, 'used': 0.0, 'total': 0.62956276}}"
2018-07-30 16:39:04,403 - freqtrade.exchange - WARNING - retrying get_balance() still for 1 times
2018-07-30 16:39:05,939 - freqtrade.exchange - WARNING - get_balance() returned exception: "Could not get USDT balance due to malformed exchange response: {'BTC': {'free': 0.62956276, 'used': 0.0, 'total': 0.62956276}}"
2018-07-30 16:39:05,939 - freqtrade.exchange - WARNING - retrying get_balance() still for 0 times
2018-07-30 16:39:07,474 - freqtrade.exchange - WARNING - get_balance() returned exception: "Could not get USDT balance due to malformed exchange response: {'BTC': {'free': 0.62956276, 'used': 0.0, 'total': 0.62956276}}"
2018-07-30 16:39:07,475 - freqtrade.exchange - WARNING - Giving up retrying: get_balance()
2018-07-30 16:39:07,475 - freqtrade.freqtradebot - WARNING - Could not get USDT balance due to malformed exchange response: {'BTC': {'free': 0.62956276, 'used': 0.0, 'total': 0.62956276}}, retrying in 30 seconds...

**Comment by vpotap** [comment]
i think i have the same issue before u need to ensure:
1. if stake currency is BTC you should have in your whitelist config.json USDT/BTC (order is important ! not BTC/USDT , if use define BTC/USDT  that's mean you have base(or stake) currency like USDT ) - in your case you should check what combination support bittrex (USDT/BTC or BTC/USDT)

2. if you set stake currency  BTC (and you don't have any BTC on real balance) only altcoins bot can't start with seil only with buy , that's why before running bot you have to ensure that stake currency available on your balance

**Comment by xmatthias** [comment]
The above explains it very well.

Stake-currency is the 2nd part of the currency pair (so for USDT/BTC , BTC is the Stake currency.

I'm closing this now, feel free to reopen if you still face issues.
